shell/platform/fuchsia/flutter/tests/fake_session_unittests.cc

```diff
@@ -0,0 +1,459 @@
+                                  VariantWith<FakeView>(FieldsAre(
+                                      view_holder_koid, view_ref_control_koid,
+                                      view_ref_koid, debug_label,
+                                      ElementsAre(Pointee(IsEntityNode(
+                                          _, node_label, IsEmpty()))),
+                                      FakeView::kDebugBoundsDisbaled))))))),
+          _, AllOf(Not(kInvalidFakeResourceId), arg.root_view_id)),
+      arg, result_listener);
+}
+
+}  // namespace
+
+class FakeSessionTest : public ::testing::Test,
+                        public fuchsia::ui::scenic::SessionListener {
+ protected:
+  FakeSessionTest()
+      : session_listener_(this), session_subloop_(loop_.StartNewLoop()) {}
+  ~FakeSessionTest() override = default;
+
+  async::TestLoop& loop() { return loop_; }
+
+  FakeSession& fake_session() { return fake_session_; }
+
+  scenic::Session CreateSession() {
+    FML_CHECK(!fake_session_.is_bound());
+    FML_CHECK(!session_listener_.is_bound());
+
+    auto [session, session_listener] =
+        fake_session_.Bind(session_subloop_->dispatcher());
+    session_listener_.Bind(std::move(session_listener));
+
+    return scenic::Session(session.Bind());
+  }
+
+ private:
+  // |fuchsia::ui::scenic::SessionListener|
+  void OnScenicError(std::string error) override { FML_CHECK(false); }
+
+  // |fuchsia::ui::scenic::SessionListener|
+  void OnScenicEvent(std::vector<fuchsia::ui::scenic::Event> events) override {
+    FML_CHECK(false);
+  }
+
+  async::TestLoop loop_;  // Must come before FIDL bindings.
+
+  fuchsia::ui::scenic::SessionPtr session_ptr_;
+  fidl::Binding<fuchsia::ui::scenic::SessionListener> session_listener_;
+
+  std::unique_ptr<async::LoopInterface> session_subloop_;
+  FakeSession fake_session_;
+};
+
+TEST_F(FakeSessionTest, Initialization) {
+  EXPECT_EQ(fake_session().debug_name(), "");
+  EXPECT_EQ(fake_session().command_queue().size(), 0u);
+  EXPECT_THAT(fake_session().SceneGraph(), IsEmptySceneGraph());
+
+  // Pump the loop one time; the session should retain its initial state.
+  loop().RunUntilIdle();
+  EXPECT_EQ(fake_session().debug_name(), "");
+  EXPECT_EQ(fake_session().command_queue().size(), 0u);
+  EXPECT_THAT(fake_session().SceneGraph(), IsEmptySceneGraph());
+}
+
+TEST_F(FakeSessionTest, DebugLabel) {
+  const std::string debug_label = GetCurrentTestName();
+  scenic::Session session = CreateSession();
+
+  // Set the session's debug name.  The `SetDebugName` hasn't been processed
+  // yet, so the session's view of the debug name is still empty.
+  session.SetDebugName(debug_label);
+  EXPECT_EQ(fake_session().debug_name(), "");
+
+  // Pump the loop; the contents of the initial `SetDebugName` should be
+  // processed.
+  loop().RunUntilIdle();
+  EXPECT_EQ(fake_session().debug_name(), debug_label);
+}
+
+TEST_F(FakeSessionTest, SimpleResourceLifecycle) {
+  const std::string node_label = "EntityNode";
```

Comment:
If you move the constant kNodeLabel down to just before first use, it would flow better.